### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by ph7jack** [wireui/wireui#897] [General]
## What's Changed
* Feature/528 v2 upgrade heroicons v1 to v2 by @PH7-Jack in https://github.com/wireui/wireui/pull/541
* Feature/V2-laravel-pint by @PH7-Jack in https://github.com/wireui/wireui/pull/547
* V2 - Refactor Card by @joaopalopes24 in https://github.com/wireui/wireui/pull/563
* Feature/v2 buttons by @PH7-Jack in https://github.com/wireui/wireui/pull/570
* Feature/v2 upgrade laravel 10 by @PH7-Jack in https://github.com/wireui/wireui/pull/654
* [v2] - Refactor Errors Component by @joaopalopes24 in https://github.com/wireui/wireui/pull/619
* [V2] Refactor Components by @joaopalopes24 in https://github.com/wireui/wireui/pull/677
* [v2] Support Livewire v3 by @joaopalopes24 in https://github.com/wireui/wireui/pull/716
* [v2] Merge V2 with Inputs by @joaopalopes24 in https://github.com/wireui/wireui/pull/722
* [v2] Notifications by @joaopalopes24 in https://github.com/wireui/wireui/pull/723
* Feature/v2 inputs by @PH7-Jack in https://github.com/wireui/wireui/pull/748
* [v2] Refactor Traits by @joaopalopes24 in https://github.com/wireui/wireui/pull/756
* [v2] Config Components by @joaopalopes24 in https://github.com/wireui/wireui/pull/755
* [v2] Adjust Packs and Props by @joaopalopes24 in https://github.com/wireui/wireui/pull/758
* [v2] Adding Unit Tests by @joaopalopes24 in https://github.com/wireui/wireui/pull/762
* [2.X] adjustments - part 1/3 by @PH7-Jack in https://github.com/wireui/wireui/pull/760
* [v2] Adding Unit Tests by @joaopalopes24 in https://github.com/wireui/wireui/pull/764
* [v2] Merge with Main by @joaopalopes24 in https://github.com/wireui/wireui/pull/772
* Bugfix of Multi-select rendering problems in Microsoft Edge by @nastoychev in https://github.com/wireui/wireui/pull/763
* Bump browserify-sign from 4.2.1 to 4.2.2 by @dependabot in https://github.com/wireui/wireui/pull/785
* V2 adjustments 2 of 3 by @PH7-Jack in https://github.com/wireui/wireui/pull/787
* [v2] Refactor Traits and Adjust Docs by @joaopalopes24 in https://github.com/wireui/wireui/pull/775
* Merge With Main by @joaopalopes24 in https://github.com/wireui/wireui/pull/790
* Fix of Issue: Error When Input Year Value at Datetime Picker #765 by @tariktokic in https://github.com/wireui/wireui/pull/786
* Added spinner option to input component. by @felipeArnold in https://github.com/wireui/wireui/pull/725
* Merge with main by @joaopalopes24 in https://github.com/wireui/wireui/pull/798
* 2.x by @PH7-Jack in https://github.com/wireui/wireui/pull/799
* [v2] Adjusts Tests by @joaopalopes24 in https://github.com/wireui/wireui/pull/801
* Remove build assets by @PH7-Jack in https://github.com/wireui/wireui/pull/805
* Removed border duplication from input when in focus. by @felipeArnold in https://github.com/wireui/wireui/pull/814
* [v2] Fix Password by @joaopalopes24 in https://github.com/wireui/wireui/pull/817
* fix getting started URL in readme.md by @codegeek1001 in https://github.com/wireui/wireui/pull/838
* Adding Component Modules by @joaopalopes24 in https://github.com/wireui/wireui/pull/858
* Laravel 11.x Compatibility by @laravel-shift in https://github.com/wireui/wireui/pull/869
* [Fix] Multi-Select Selecting Issue by @devsquad-gabriel-ramos in https://github.com/wireui/wireui/pull/861
* Create messages.php for Turkish language by @vedatyilmaz in https://github.com/wireui/wireui/pull/806
* added rows and cols attributes to the textarea by @NathanBeddoeWebDev in https://github.com/wireui/wireui/pull/853
* [V2]: Adjust Version 2 by @joaopalopes24 in https://github.com/wireui/wireui/pull/894
* Fix time picker by @PH7-Jack in https://github.com/wireui/wireui/pull/896

## New Contributors
* @nastoychev made their first contribution in https://github.com/wireui/wireui/pull/763
* @tariktokic made their first contribution in https://github.com/wireui/wireui/pull/786
* @felipeArnold made their first contribution in https://github.com/wireui/wireui/pull/725
* @codegeek1001 made their first contribution in https://github.com/wireui/wireui/pull/838
* @laravel-shift made their first contribution in https://github.com/wireui/wireui/pull/869
* @devsquad-gabriel-ramos made their first contribution in https://github.com/wireui/wireui/pull/861
* @vedatyilmaz made their first contribution in https://github.com/wireui/wireui/pull/806
* @NathanBeddoeWebDev made their first contribution in https://github.com/wireui/wireui/pull/853

**Full Changelog**: https://github.com/wireui/wireui/compare/v1.18.9...v2.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/wireui/wireui/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by madushancs** [comment]
Hi. Thank you for hard work. Any update instruction. I have v1.19.1

**Reply by cbaconnier** [reply]
So far I have seen:

**tailwind.config.js** : Update the paths of *content*
```diff
-'./vendor/wireui/wireui/resources/**/*.blade.php',
-'./vendor/wireui/wireui/ts/**/*.ts',
-'./vendor/wireui/wireui/src/View/**/*.php'
+"./vendor/wireui/wireui/src/*.php",
+"./vendor/wireui/wireui/ts/**/*.ts",
+"./vendor/wireui/wireui/src/WireUi/**/*.php",
+"./vendor/wireui/wireui/src/Components/**/*.php",
```

If you uses dialogues you have to update your Livewire classes using 

```diff
use Livewire\Component;
- use WireUi\Traits\Actions;
+ use WireUi\Traits\WireUiActions;

class MyComponent extends Component
{
-    use Actions;
+    use WireUiActions;
```

*Not sure about this one. I may did something odd at the first place ?*
```diff
- <x-inputs.currency
+ <x-currency
```

All the icons has been changed to [heroicons](https://heroicons.com/)
```diff
- <x-icon name="download" />
+  <x-icon name="arrow-down-tray" /> 

- <x-icon name="upload" />
+  <x-icon name="arrow-up-tray" /> 

- <x-icon name="upload" />
+  <x-icon name="arrow-up-tray" /> 

- <x-input icon="search" .../>
+ <x-input icon="magnifying-glass" .../>

...
```
I believe there is more do this, I only checked what I use. Anyone feel free to complete the list

**Reply by joaopalopes24** [reply]
We are finalizing the Upgrade Guide in the documentation. We will go up this week.

**Reply by luigiaureagroup** [reply]
> We are finalizing the Upgrade Guide in the documentation. We will go up this week.

Hi, has it already been published?

**Reply by kennyelliott** [reply]
I'm anxious to see the upgrade guide also.

**Reply by kennyelliott** [reply]
Any update on the upgrade guide?

**Reply by ypho** [reply]
@joaopalopes24 any news on the update guide? I'd love to go to v2.

**Reply by dwainscheeren** [reply]
The upgrade guide is taking long..

**Reply by Luigidefra** [reply]
> We are finalizing the Upgrade Guide in the documentation. We will go up this week.

Hi, how's it going, it's been over a week already :)
I'm using v1 but need features that are on v2.
I'm stuck and I need the Upgrade Guide, could you give me further information on the status of the publication?

Thanks so much for everything!

**Reply by dwainscheeren** [reply]
I don't think we are getting the upgrade guide unfortunately
